### ISSUE TITLE
Links/References all have the type `Link` which makes them not distinguishable

### ISSUE THREAD

**Issue by brokelyn** [contentful/ui-extensions-sdk#1385]
I try to build a Contentful App using the `@contentful/default-field-editors`. I am using the `FieldExtensionSDK` to figure out the field type by looking at `sdk.field` which is an object of the class [FieldAPI](https://github.com/contentful/ui-extensions-sdk/blob/master/lib/types/field-locale.types.ts). Regarding to this [source](https://www.contentful.com/developers/docs/references/content-management-api/#/reference/ui-extensions/extensions-collection) the field type can be one out of these types:

- {type: "Symbol"}
- {type: "Text"}
- {type: "RichText"}
- {type: "Integer"}
- {type: "Number"}
- {type: "Date"}
- {type: "Boolean"}
- {type: "Location"}
- {type: "Object"}
- {type: "Link", linkType: "Asset"}
- {type: "Link", linkType: "Entry"}
- {type: "Array", items: {type: "Symbol"}}
- {type: "Array", items: {type: "Link", linkType: "Entry"}}
- {type: "Array", items: {type: "Link", linkType: "Asset"}}

This works correct for all field types except for Links and Arrays (the last 5 items from the list). For any link whether it is `{type: "Link", linkType: "Asset"}`, `{type: "Array", items: {type: "Symbol"}}` or one of the other 3 reference types the `FieldAPI` object I receive looks like:

```
{ id: ..., locale: ..., required: ..., type: "Link", items: undefined, ... }
```

Since the `type` field is always `Link` and `items` field is always `undefined` it is not possible to render the correct default field editor for this field. Though it seems to be possible to fetch the more detailed information from the API (mentioned above) and the `default-field-editor` supports these different types aswell, [see here](https://github.com/contentful/field-editors/blob/master/packages/default-field-editors/src/types.ts#L24).

So my question is if this is a bug or maybe this is a feature which can be implemented? Maybe the information is somewhere hidden and I do not see it? I am happy to provide more information if needed.

Package versions
```
"@contentful/app-sdk": "^4.12.1",
"@contentful/default-field-editors": "^1.2.38",
"@contentful/f36-components": "^4.21.0",
"contentful-management": "^10.18.1",
"react": "^16.14.0",
```

**Comment by martin3walker** [comment]
Hey Philipp,

Is the problem that you're unable to store values in the [default field editors](https://contentful-field-editors.netlify.app/) for references? 

I'm a bit confused because it sounds like you're trying to render a field editor based on the data that is stored in the entry, but I think it should go the other way around. You're providing editors with one of the default field editors, so they can store a certain type of data. 

For instance, if you want a field that stores images, then you could render the `MultipleMediaEditor` (https://contentful-field-editors.netlify.app/media-multiple) to allow editors to store images. Is that helpful?

**Comment by brokelyn** [comment]
No not really. I will try to make a small example of what I try to achive or where my problem is.

I have Content model `A` which has only one field called `MediaFiles`. This field is of type `Media` with `Many files` selected. As appearance I choose my own App.

When I now try to create an instance/object of this Content model `A` I expect my App to use the `default-field-editor` package and render a `MultipleMediaEditor` since the field `MediaFiles` is of type `Media` and can contain multiple files.

Unfortunately the `default-field-editor` wont render anything since it cant determine the type of the field `MediaFiles`. Looking at the `sdk` object I pass to `default-field-editor` I can see that the type I get from the `FieldExtensionSDK` (part of the App SDK) is looking like this:

```
{ id: ..., locale: ..., required: ..., type: "Link", items: undefined, ... }
```

but I would expect it to look like this (more information so I can render the correct component):

```
{ id: ..., locale: ..., required: ..., type: "Array", items: {type: "Link", linkType: "Asset"}, ... }
```

So basically what is described [here](https://contentful-field-editors.netlify.app/shared/default-field-editors), just passing the `FieldExtensionSdk` and the `default-field-editor` figures out which type of field it needs to render is not working for references.

I hope now it is a bit more clear. Let me know if not!

**Comment by fkoschi** [comment]
@brokelyn I do have the exact same problem right now.

**Comment by brokelyn** [comment]
Good to see that I am not alone. At the moment I have a focus on something else but I probaly come back to this and investigate further. 
I also saw this described in some more pages in the docs now, so it is definitely possible!

**Comment by brokelyn** [comment]
I did some more research today. I tracked down the source of the data coming from the code below (which is in the `channel.ts` file). 
```
function waitForConnect(currentGlobal: typeof globalThis, onConnect: Function) {
  currentGlobal.addEventListener('message', listener)

  function listener(event: MessageEvent) {
    const message = event.data
    if (message.method === 'connect') {
      console.log(event.data)
      currentGlobal.removeEventListener('message', listener)
      onConnect(...message.params)
    }
  }
}
```
I assume the data is injected by the Contentful website and therefore the App-SDK is not directly fetching the data from the API. Is my assumption correct?

Currently for my test setup I have two `Content models`. Turns out for the first of the `Content Model` the above described error appears but for the second `Content Model` everything works as expected. 

But in addition I cant set my app as appearance on the second `Content Model` for fields that have multiple references (like `Many References` or `Many Media Assets`). In the appearance tab my app just does not show up. For the first `Content Model` though it still works.

 Is there currently an issue with setting the `Appearance` for `Many References` and `Many Media Assets`?

would be nice if you could check this out again @martin3walker

**Comment by brokelyn** [comment]
It seems to be a big confusion here since in the docs it is described how you can check for the `field.type` and distinguish between `Array` and normal `Link` but it seems like that you cannot set the `Appearance` to your custom app on `Many References` or `Many Media Assets`.

@fkoschi I guess I saw your `appearance applied` somewhere in the raw data coming from Contentful here in the `channel.ts` file. So it might be possible to add it to the App-SDK
```
function waitForConnect(currentGlobal: typeof globalThis, onConnect: Function) {
  currentGlobal.addEventListener('message', listener)

  function listener(event: MessageEvent) {
    const message = event.data
    if (message.method === 'connect') {
      console.log(event.data)
      currentGlobal.removeEventListener('message', listener)
      onConnect(...message.params)
    }
  }
}
```

sorry for the confusion
